pkg/graph/resolvers/mto_solution.go

```diff
@@ -0,0 +1,150 @@
+package resolvers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+	"go.uber.org/zap"
+
+	"github.com/cms-enterprise/mint-app/pkg/authentication"
+	"github.com/cms-enterprise/mint-app/pkg/models"
+	"github.com/cms-enterprise/mint-app/pkg/sqlutils"
+	"github.com/cms-enterprise/mint-app/pkg/storage"
+	"github.com/cms-enterprise/mint-app/pkg/storage/loaders"
+)
+
+// MTOSolutionGetByModelPlanIDLOADER implements resolver logic to get all MTO
+// solutions by a model plan ID using a data loader
+func MTOSolutionGetByModelPlanIDLOADER(
+	ctx context.Context,
+	modelPlanID uuid.UUID,
+) ([]*models.MTOCommonSolution, error) {
+
+	// Translate a nil key to UUID nil, as we need a primitive type for translating results later
+	return loaders.MTOCommonSolution.ByModelPlanID.Load(ctx, modelPlanID)
+}
+
+func MTOSolutionGetCommonSolutionByKeyLoader(
+	ctx context.Context,
+	namedPreparer sqlutils.NamedPreparer,
+	logger *zap.Logger,
+	key *models.MTOCommonSolutionKey,
+) (*models.MTOCommonSolution, error) {
+	if key == nil {
+		return nil, fmt.Errorf("common solution key is nil")
```

Comment:
We might need to look at this for both this code and for Milestones, but my guess is that we don't want to throw an error if we try and fetch a Common Solution for a given MTO Solution, but would rather just return `nil, nil`, since it's not really an error if you don't have a key, it just means you don't also have a common solution (which the return type is a pointer, so we can probably just modify this code as such)


```suggestion
		return nil, nil
```

Reply:
https://github.com/CMS-Enterprise/mint-app/blob/c7dbfa452a1f97c6d9b127c28cfaccfb8f7868bf/pkg/graph/resolvers/mto_milestone.resolvers.go#L18-L24

Actually, there's this code on the Milestones that handles this -- might just be worth adding that to the resolver for this one (not in this file, but in `mto_solution.resolvers.go`, similar to how the change I linked is in `mto_milestone.resolvers.go`)